### DISCUSSION TITLE
Serial Port

### DISCUSSION THREAD

**Discussion by ghost** [ltsp/ltsp#393] [Q&A] [answered]
Hi, I wrote a program that uses the serial port(ttyUSB0), it works fines when executed from a local Linux installation, but when I try to use it from the same image served from ltsp, it no longer works, do i need to do some extra configuration for the serial port to work on the client machine  ? The user I'm using to execute it (its the same on the server and the client) is a member of the dialout group. I read something about local apps, but it appears to be deprecated.

Thanks.

**Comment by alkisg** [comment]
Hi, what's the output of the following commands on the client, when connected as the user?

```shell
id
ls -l /dev/ttyUSB0
ltsp info
```

**Comment by ghost** [comment]
Hi alkisg, thanks for you help again, here is the output  : 
id :
uid=1000(pharaoh) gid=1000(pharaoh) groups=1000(pharaoh),4(adm),20(dialout),24(cdrom),27(sudo),30(dip),46(plugdev),121(lpadmin),131(lxd),132(sambashare),133(epoptes)

ls -l /dev/ttyUSB0 :
crw-rw---- 1 root dialout 188, 0 fev 2 14:39 /dev/ttyUSB0

ltsp info : 
SERVER XSESSIONS: plasma.desktop
SERVER OS: PRETTY_NAME="Ubuntu 20.04 LTS"
CHROOTS:
VMs:
IMAGES:

**Comment by alkisg** [comment, marked as answer]
Everything seems OK, that user seems to have read/write access to /dev/ttyUSB0.
I can't think of anything that LTSP does that would affect its access to the serial port.

Try to run the program from a terminal and see if it shows any useful error messages in its stderr.
Try also to boot that specific client with a live CD and to run the program there.

**Comment by ghost** [comment]
Hi alkisg, you were right, the problem was a mistake i did.

Thanks again for your support.
